### PR TITLE
On-click buff timers and typing indicator

### PR DESCRIPTION
**Pull request by JohnWildkins**
Implements a typing indicator 
image

Implements buff timers when buff icons are clicked
image

### CODE REVIEW

File: SWLOR.Game.Server/Feature/BuffTimer.cs [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+            if (!GetIsPC(player)) return;
+            if (type != GuiEventType.EffectIconClick) return;
+
+            var buffInt = GetLastGuiEventInteger();
+
+            // We don't care about duration for concentration powers, because their duration is arbitrary
+
+            switch(buffInt)
+            {
+                case (int)EffectIconType.Slow:
```
Comment (zunath) [inline SWLOR.Game.Server/Feature/BuffTimer.cs:30]: This is going to be hard to maintain as we add more effects or change the meaning around the icons. I'd recommend just pulling the name off the 2DA file, should be a one-liner instead.

File: SWLOR.Game.Server/Feature/GuiDefinition/PerksDefinition.cs [hunk 2]
```diff
@@ -98,7 +98,7 @@ public GuiConstructedWindow BuildWindow()
                             });
                         })
                             .SetRowHeight(40f)
-                            .SetScrollbars(NuiScrollbars.Both)
+                            .SetScrollbars(NuiScrollbars.Y)
```
Comment (zunath) [inline SWLOR.Game.Server/Feature/GuiDefinition/PerksDefinition.cs:101]: What are the changes to the perk window here?

File: SWLOR.Game.Server/Service/Communication.cs [hunk 3]
```diff
@@ -77,8 +77,30 @@ public static void LoadHolonetSetting()
...
+        [NWNEventHandler("mod_gui_event")]
+        public static void TypingIndicator()
+        {
+            var player = GetLastGuiEventPlayer();
+            var type = GetLastGuiEventType();
+            if (!GetIsPC(player)) return;
+
+            if(type == GuiEventType.ChatBarFocus)
+            {
+                var chatIndic = TagEffect(EffectVisualEffect(Core.NWScript.Enum.VisualEffect.VisualEffect.Vfx_Dur_Chat_Bubble, false, 0.5f), "typingindicator");
```
Comment (zunath) [inline SWLOR.Game.Server/Service/Communication.cs:94]: Throw a using statement for the VisualEffect namespace on this file please. Easier to read without the fully qualified name.

**Review by zunath — CHANGES_REQUESTED** [review]
Requesting changes and answers to comments above.

File: SWLOR.Game.Server/Feature/GuiDefinition/PerksDefinition.cs (hunk 2 above)
Comment (JohnWildkins) [reply SWLOR.Game.Server/Feature/GuiDefinition/PerksDefinition.cs:101]: An oopsie, I split these timers out from a larger PR and missed this file, mb

File: SWLOR.Game.Server/Feature/GuiDefinition/PerksDefinition.cs [hunk 4]
```diff
@@ -98,7 +98,7 @@ public GuiConstructedWindow BuildWindow()
                             });
                         })
                             .SetRowHeight(40f)
-                            .SetScrollbars(NuiScrollbars.Y)
+                            .SetScrollbars(NuiScrollbars.Both)
```
Comment (zunath) [inline SWLOR.Game.Server/Feature/GuiDefinition/PerksDefinition.cs:101]: Your perk window changes came back. Intended? What do they do?

File: SWLOR.Game.Server/Feature/ChatCommandDefinition/CharacterChatCommand.cs [hunk 5]
```diff
@@ -326,5 +327,25 @@ private void Customize()
...
+
+        private void AlwaysWalk()
+        {
+            _builder.Create("alwayswalk", "walk")
+                .Description("Forces walking when moving your character. \"/alwayswalk off\" or \"/walk off\" to disable.")
+                .Permissions(AuthorizationLevel.All)
+                .Action((user, _, _, args) =>
+                {
+                    if (args.Length > 0 &&
+                        (args[0] == "off" || args[0] == "0"))
```
Comment (zunath) [inline SWLOR.Game.Server/Feature/ChatCommandDefinition/CharacterChatCommand.cs:339]: Spanish guys gonna be annoyed with this one :P

**Review by zunath — APPROVED** [review]

**Pull request merged**